Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -1 +1,894 @@
+CREATE TABLE my_table (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100) NOT NULL
+);
+
+-- Add 2 numbers using add function
+CREATE FUNCTION add(a integer, b integer) RETURNS integer
+    LANGUAGE SQL
+    IMMUTABLE
+    RETURNS NULL ON NULL INPUT
+    RETURN a + b;
+```
+
+### Step 4: Create the Makefile
+- Create a Makefile in the "extension" directory. 
+  - The Makefile is used to define the build process. It specifies the module name, extension name, and SQL files. The PG_CONFIG variable retrieves the path to the pg_config utility, which is used to get the necessary configuration for building the extension.
+
+```bash
+$ touch Makefile
+```
+- Open the Makefile and add the following content:
+
+```makefile
+EXTENSION = my_extension                 # Name of the extension
+DATA = my_extension--1.0.0.sql           # SQL file containing extension objects and functions
+
+PG_CONFIG  ?= pg_config                  # Path to the pg_config executable
+PGXS := $(shell $(PG_CONFIG) --pgxs)     # Get the PostgreSQL Extension Makefile from pg_config
+include $(PGXS)                          # Include the PostgreSQL Extension Makefile
+```
+
+<!--
+We could explain PGXS inshort to give the user more idea about the last 3 lines in the makefile.
+#### PGXS (PostgreSQL Extension Build Infrastructure) 
+It is a framework that helps simplify the process of building and installing extensions for PostgreSQL. It provides a standard way to compile and install extensions, making it easier to distribute and use them.
+
+To use PGXS:
+1. Create your extension's files, including source code and control files.
+2. Create a Makefile in your extension's directory.
+3. Inside the Makefile, include PGXS by adding a few lines of code.
+4. Define any extension-specific instructions in the Makefile.
+5. Use the `make` command to build and install your extension.
+6. Enable the extension in PostgreSQL using the `CREATE EXTENSION` SQL command.
+
+PGXS takes care of locating your PostgreSQL installation, determining compilation options, and ensuring the extension files are in the right place. It simplifies the process, making it easier for you to build and install your extension in different PostgreSQL environments.
+-->
+
+For more information on the Makefiles, you can refer to the [PostgreSQL documentation on makefile](https://www.postgresql.org/docs/15/extend-pgxs.html).
+
+#### Tip: 
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.0.sql my_extension--1.1.0.sql
+```
+
+- By including multiple SQL script files in the `DATA` variable of the `Makefile`, the user can ensure that different versions of the extension can have different sets of objects or functionality. 
+- Each SQL script file represents a specific version of the extension and can contain the necessary SQL commands for the desired changes.
+- This allows the user to easily manage and control the installation of specific features or modifications in different versions of your extension.
```

Comment:
It took me a few minutes to parse this section. I think the important point is that different versions of an extension can be supported by having files for each version, e.g., `my_extension-1.0.0.sql`, `my_extension-1.0.1.sql`, `my_extension-1.2.0.sql`, etc. You tell Postgres about them by including them all in the `DATA` variable, either as a list or using wildcards, e.g., `DATA = $(wildcard my-extension--*.sql)`.

Reply:
I was suggested this by a fellow gsoc contributor when i asked him for some feedback. He told me that he had a problem with understanding this part. Most people know about the wildcard but including the 2-3 SQL scripts to load all of them isn't mentioned probably.

I can add a section about wildcard to give a better approach for having all the versions.

Reply:
I think it's fine to explicitly list the files as an example; my point is that I had a hard time understanding your explanation.

Reply:
ohk, i will change it up and submit it for review again.